Video/tormyvancool_YOUTUBE Downloader.lua

```diff
@@ -0,0 +1,367 @@
+          if string.find(FileName, ".mp4") == nil then
+            FileTemp = FileName .. '.f137.mp4.part'
+            FileName = FileName .. ".mp4"
+          end
+          argument = ' -o "'  .. FileName .. '"'
+          FileTemp = FileName:sub(1, -5) .. '.f137.mp4.part'
+      end
+      
+      -- ARGS
+      args = " --update-to master -S vcodec:h264,res,acodec:aac " .. url .. ' -P "' .. ProjDir .. '/Videos/"' .. argument .. " --force-overwrite"
+      upArgs = " --update-to master"
+
+      -- TRIGGERS
+      Update = Start .. MainPath .. upArgs
+      Video = Start .. MainPath .. args
+      Destination =  ProjDir ..'/Videos/' .. FileName
+      Destination = Destination:gsub('\\','/')
+      
+
+--cd ~/Library/"Application Support"/REAPER/Scripts/Tormy\ Van\ Cool\ ReaPack\ Scripts/Various/yt-dlp/ && ./yt-dlp_macos
+
+---------------------------------------------
+-- UPDATE AND IMPORT VIDEO
+---------------------------------------------
+
+      if url  ~= "" then
+
+          if debug == true then 
+            reaper.ShowConsoleMsg("FileName: " .. FileTemp .. "\n")
+            reaper.ShowConsoleMsg("Destination: " .. Destination .. "\n")
+          end
+          
+          -- CHECK IF FILE EXISTS
+          local checkfile = reaper.file_exists(Destination)
+          local answer = nil
+          if checkfile == true then
+            answer = reaper.MB("A file with the same filename exists\nWould you want to overwrite it?\n\nYES => Go on\nNO => Rewrite the filname\nCANCEL => Exit",'WARNING: FILENAME EXISTS',3)
+          end
+          if answer == 7 then 
+            goto getfilename 
+          elseif answer == 2 then
+            goto done
+          end
+          
+          reaper.MB("STARTED THE FOLLOWING PROCESSES v" .. ver .. ":\n\n1. Update YT-DLP\n2. Downlaod the video: " ..url .. "\n3. Naming the video: " .. FileName .. ".mp4 \n4. Saving the video into " .. ProjDir .. "/Videos/\n5. Import the video into the project\n\nHEY it will take a little while. DON'T PANIC!\n\nCLICK ON \"OK\" TO CONTINUE", "PROCESS STARTED. PROCESSES LISTED HERE BELOW",0)
+          
+          -- DIFFERENTIATE EXECUTION BASED ON O.S.
+          if OpSys == 2 then
+            os.execute(Update)
+            os.execute(Video)
+          elseif OpSys == 1 or OpSys == 3 then
+            os.execute(Start .. MainPath .. upArgs .. args)
+          end
+          
+          
+          ---------------------------------------------
+          -- NETWORK DISRUPTION
+          ---------------------------------------------
+          
+          -- GET RESIDUAL FILES
+          local ResFiles
+          if OpSys == 2 or OpSys == 3 then
+              ResFiles = "ls -a " .. ProjDir .."/Videos/*.mp4.part"
+              ResDel = 'rm "' .. ProjDir .. '\\Videos\\*.mp4.part"'
+          elseif OpSys == 1 then
+              ResFiles = 'dir /b "' .. ProjDir .. '\\Videos\\*.mp4.part"'
+              ResDel = 'del "' .. ProjDir .. '\\Videos\\*.mp4.part"' 
```

Comment:
There are API functions for this that don't require going through system commands. Also (at least) the Unix code path is not only dangerous but also broken: the star within quotes is interpreted literally not as a globbing operator.

Reply:
I tried that command in Linux virtual machine (Ubuntu) as it is, before writing it into this  code and it works.
About the APIs you are referring to: which are?
Are they doing exactly what I want from these lines. The goal is to remove any `.mp4.part` file that's forming inside that specific subfolder.
I I knew they existed I could use them; APIs are making my life easier (when and IF correctly explained)

Reply:
yt-dlp does not remove these files self when the network fails. No way.
These commands are entering in actions ONLY if at the end of the process, yt-dlp failed/crashed and .mp4.part will still there forever as leftovers, otherwise.

Instead if yt-dlp works fine, they do not remove anything because there is nothing to be removed.

Reply:
There's also no way `rm "/reaper/resource/path\Videos\*.mp4.part"` does anything beside output `No such file or directory` to stderr...

Reply:
I'm checking what I can do to get the identical thing I want, by the use of the suggested APIs

and for the "rm" not clue why it worked on my virtual machine then.

Reply:
NOTE in `integer fileindex` it will be nice to understand what's expected. I just used the -1 only

I supposed it should be used in this way  `ResFiles = reaper.EnumerateFiles( ProjDir .. '/Videos/*.mp4.part', 1)`
But unlike the commands I wrote and they are working, this one doesn't :-/

Reply:
`reaper.EnumerateFiles(ProjDir .. '/Videos', i)` in a loop with i incremented by one each iteration starting from 0 until nil is returned. Check if the returned filename ends with '.mp4.part' and delete if so...

Reply:
Ok fodn the way

```
              a=0
              test = ""
              repeat
              reaper.ShowConsoleMsg(test)
              test = reaper.EnumerateFiles( ProjDir .. "\\Videos\\*.mp4.part", a)
              a = a + 1
              until(test == nil)
```

I will adapt it and implement it

Reply:
Sorry I read you after have posted mien one. Ok indeed I'm doing that implementation and some part of the code mus be rewritten although it will be simplified and more efficient

Reply:
Done, committed